src/components/tasks/task-dialog.tsx

```diff
@@ -121,6 +159,121 @@ const TaskDialog = ({ projectId, milestoneId, open, task, onClose }: Props) => {
     onError: (error) => console.error(t("errorHandling.errorUpdatingMilestoneTask"), error),
   });
 
+  /**
+   * Create task connections mutation
+   */
+  const createTaskConnectionsMutation = useMutation({
+    mutationFn: (params: CreateTaskConnectionRequest) => taskConnectionsApi.createTaskConnection(params),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["taskConnections", projectId, task?.id] });
```

Comment:
Should milestone id also be included in the query key here?

Reply:
There are a few best practices for proper query keys.

1. Query Keys should _always_ include everything that the query depends on
2. Query keys should have proper hierarchy, so that we can use query invalidations effectively.
3. If there are parameters that filter the results from the query, they should be placed as properties to a single object. The object should be in placed the query key array based on the hierarchy. Usually this means the last place.

As such, IMO one great option to implement proper query keys is to use actual API endpoint routes, since those are already quite thought out from the relations standpoint. For example, the key in this query would then be `["projects", projectId, "connections", { taskId: task?.id }]`.

This means, that if we want to invalidate connections for a specific task id, you would invalidate it by providing that exact query key. If we want to invalidate ALL of the connections, we would just use `["projects", projectId, "connections"]`. If we would also want to invalidate all the queries related to the project (including the connections), we would use `["projects", projectId]`, and so on.

Reply:
Here is also a pretty good blog post about effective query keys:
https://tkdodo.eu/blog/effective-react-query-keys

Reply:
Good explanation, thank you